full_test.py

```diff
@@ -4,49 +4,77 @@
 from io import open
 
 from pyuca import Collator
+
+from pyuca.collator import (
+    Collator_5_2_0, Collator_6_3_0,
+    Collator_8_0_0, Collator_9_0_0, Collator_10_0_0
+)
 from pyuca.utils import format_sort_key
 
 try:
     chr = unichr
 except NameError:
     pass
 
-c = Collator()
-
-prev_sort_key = None
-
-success = 0
-failure = 0
-
-path = "CollationTest/{0}/CollationTest_NON_IGNORABLE.txt".format(
-    c.UCA_VERSION)
-
-with open(path) as f:
-    for i, line in enumerate(f.readlines()):
-        points = line.split("#", 1)[0].split(";", 1)[0].strip().split()
-
-        if points:
-            test_string = "".join(
-                chr(int(point, 16)) for point in points
-            )
-            test_string_sort_key = c.sort_key(test_string)
-            if prev_sort_key:
-                if prev_sort_key > test_string_sort_key:
-                    failure += 1
-                    print('-------')
-                    print("failed on line {0}:".format(i+1))
-                    print(line.rstrip('\n'))
-                    print("PREV: {0}".format(format_sort_key(prev_sort_key)))
-                    print("THIS: {0}".format(
-                        format_sort_key(test_string_sort_key)))
-                    print('-------')
-                else:
-                    success += 1
-            prev_sort_key = test_string_sort_key
-
-print("")
-print("{0} success; {1} failure (UCA version {2})".format(
-    success, failure, c.UCA_VERSION))
-
-if failure > 0:
+PYTHON3 = sys.version_info >= (3,)
+V8_0_0 = sys.version_info >= (3, 5)
+V9_0_0 = sys.version_info >= (3, 6)
+V10_0_0 = sys.version_info >= (3, 7)
+
+collators = [Collator, Collator_5_2_0]
```

Comment:
This will cause default collator test exectured twice. For example. my python version is 
```
>>> sys.version_info
sys.version_info(major=3, minor=5, micro=3, releaselevel='final', serial=0)
```

And I get this when running test:

```
pyuca.git (master)$ python3 full_test.py
191537 success; 0 failure (UCA version 8.0.0)
152853 success; 0 failure (UCA version 5.2.0)
166270 success; 0 failure (UCA version 6.3.0)
191537 success; 0 failure (UCA version 8.0.0)
```

I suppose that line should be
```python
collators = [Collator_5_2_0]
```

Reply:
yes, it calls it twice but if you just had

```collators = [Collator_5_2_0]```

then the logic for dispatching to the default collator wouldn't be tested.

I figured running a test twice wasn't that big a deal.

Reply:
I came up with a quick solution